src/solvers/dg_simplices/dg.jl

```diff
@@ -98,43 +104,43 @@ function create_cache(mesh::VertexMappedMesh, equations, dg::DG, RealT, uEltype)
   flux_face_values = similar(u_face_values)
 
   # local storage for fluxes
-  flux_values = StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq), nvars))
+  flux_values_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq), nvars)) for _ in 1:Threads.nthreads()]
   
-  return (; md, invMQrTrW, invMQsTrW, invJ = inv.(md.J),
-      u_values, flux_values, u_face_values, flux_face_values)
+  return (; md, invMQrstTrW, invJ = inv.(md.J),
+      u_values, flux_values_threaded, u_face_values, flux_face_values)
 end
 
 function calc_volume_integral!(du,u::StructArray, volume_integral::VolumeIntegralWeakForm,
-                 mesh::VertexMappedMesh, equations, dg::DG{<:RefElemData{2}}, cache)
+                 mesh::VertexMappedMesh, equations, dg::DG{<:RefElemData{Dim}}, cache) where {Dim}
 
   rd = dg.basis  
   md = mesh.md
-  @unpack invMQrTrW, invMQsTrW, u_values, flux_values = cache
-  @unpack rxJ, sxJ, ryJ, syJ = md # geometric terms
+  @unpack invMQrstTrW, u_values, flux_values_threaded = cache
+  @unpack rstxyzJ = md # geometric terms
 
   # interpolate to quadrature points
   StructArrays.foreachfield(mul_by!(rd.Vq), u_values, u)
 
   # Todo: simplices. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
   @threaded for e in eachelement(mesh, dg, cache)
-    flux_values .= flux.(view(u_values,:,e), 1, equations)
-    StructArrays.foreachfield(mul_by_accum!(invMQrTrW, rxJ[1,e]), view(du,:,e), flux_values)
-    StructArrays.foreachfield(mul_by_accum!(invMQsTrW, sxJ[1,e]), view(du,:,e), flux_values)
-
-    flux_values .= flux.(view(u_values,:,e),2,equations)
-    StructArrays.foreachfield(mul_by_accum!(invMQrTrW, ryJ[1,e]), view(du,:,e), flux_values)
-    StructArrays.foreachfield(mul_by_accum!(invMQsTrW, syJ[1,e]), view(du,:,e), flux_values)
+    
+    flux_values = flux_values_threaded[Threads.threadid()]
+    for i in eachdim(mesh, dg, cache) 
+      flux_values .= flux.(view(u_values,:,e), i, equations)
+      for j in eachdim(mesh, dg, cache)
+        StructArrays.foreachfield(mul_by_accum!(invMQrstTrW[j], rstxyzJ[i,j][1,e]), 
+                                  view(du,:,e), flux_values)
+      end
+    end
```

Comment:
Would need to be adapted as well if you apply my suggestion above
```suggestion
    for i in eachdim(mesh) 
      flux_values .= flux.(view(u_values,:,e), i, equations)
      for j in eachdim(mesh)
        StructArrays.foreachfield(mul_by_accum!(invMQrstTrW[j], rstxyzJ[i,j][1,e]), 
                                  view(du,:,e), flux_values)
      end
    end
```